chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/developer/attestation/DeveloperAttestationReportDataCollection.java

```diff
@@ -0,0 +1,358 @@
+
+        return reportRows;
+    }
+
+    private Boolean doesActiveListingExistDuringAttestationPeriod(List<CertifiedProductBasicSearchResult> listingsForDeveloper, AttestationPeriod period) {
+        return listingsForDeveloper.stream()
+                .filter(listing -> isListingActiveDuringPeriod(listing, period))
+                .findAny()
+                .isPresent();
+    }
+
+    private Boolean isListingActiveDuringPeriod(CertifiedProductBasicSearchResult listing, AttestationPeriod period) {
+        List<CertificationStatusEvent> statusEvents = listing.getStatusEvents().stream()
+                .map(x ->  CertificationStatusEvent.builder()
+                        .status(CertificationStatus.builder()
+                                .name(x.split(":")[1])
+                                .build())
+                        .eventDate(toDate(LocalDate.parse(x.split(":")[0])).getTime())
+                        .build())
+                .sorted(Comparator.comparing(CertificationStatusEvent::getEventDate))
+                .toList();
+
+        return isListingActiveDuringAttestationPeriod(statusEvents, period);
+    }
+
+    private List<CertifiedProductBasicSearchResult> getListingDataForDeveloper(Developer developer) {
+        if (!developerListings.containsKey(developer.getDeveloperId())) {
+            SearchRequest request = SearchRequest.builder()
+                    .certificationEditions(Stream.of(CertificationEditionConcept.CERTIFICATION_EDITION_2015.getYear()).collect(Collectors.toSet()))
+                    .developer(developer.getName())
+                    .pageSize(MAX_PAGE_SIZE)
+                    .build();
+
+            try {
+                SearchResponse response = listingSearchService.search(request);
+                developerListings.put(developer.getDeveloperId(), response.getResults());
+            } catch (ValidationException e) {
+                LOGGER.error("Could not retrieve listings for developer {}.", developer.getName());
+                LOGGER.error(e);
+                developerListings.put(developer.getDeveloperId(), new ArrayList<CertifiedProductBasicSearchResult>());
+            }
+        }
+        return developerListings.get(developer.getDeveloperId());
+    }
+
+    private List<Developer> getAllDevelopers() {
+        return developerDAO.findAll().stream()
+                .map(dto -> new Developer(dto))
+                .toList();
+    }
+
+    private DeveloperAttestationSubmission getDeveloperAttestation(Long developerId, Long attestationPeriodId) {
+        List<DeveloperAttestationSubmission> attestations =
+                attestationDAO.getDeveloperAttestationSubmissionsByDeveloperAndPeriod(developerId, attestationPeriodId);
+
+        if (attestations != null && attestations.size() > 0) {
+            return attestations.get(0);
+        } else {
+            return null;
+        }
+    }
+
+    //TODO - This s/b refactored to use AttestationPeriodService
```

Comment:
Still todo? Leftover comment?

Reply:
This service was created in OCD-3790 and is not yet on STG.  If I need to make a change after OCD-3790 goes to STG, I'll update this, otherwise, it's a note for next time this report gets cracked open.